gradle-incremental-configuration-cache/src/main/java/com/palantir/gradle/configcache/incremental/DryRunConfigurationCacheEnabledTask.java

```diff
@@ -0,0 +1,210 @@
+
+                    (Failed to save CircleCI artifact: %s)
+                    """,
+                    buildDetailedErrorMessage(output, Optional.empty()), e.getMessage());
+        }
+    }
+
+    @SuppressWarnings("checkstyle:LineLength")
+    private String buildDetailedErrorMessage(Optional<String> outputContent, Optional<String> validationReportUrl) {
+        Optional<String> configCacheReportPath = extractConfigCacheReportPath(outputContent);
+
+        // Get the proper artifact URL for the configuration cache report
+        Optional<String> configCacheReportUrl = Optional.empty();
+        if (configCacheReportPath.isPresent() && validationReportUrl.isPresent()) {
+            configCacheReportUrl = configCacheReportArtifactUrl(configCacheReportPath.get());
+        }
+
+        // Build the reports section based on what's available
+        String reportsSection = reportsSection(validationReportUrl, configCacheReportUrl, configCacheReportPath);
+
+        // Build the main message (shared between CI and local)
+        String mainMessage = String.format(
+                """
+            ❌ CONFIGURATION CACHE ALLOW LIST VALIDATION FAILED
+
+            WHAT HAPPENED:
+              Some task / tasks in the allow list failed to run with configuration cache enabled.
+
+            %s
+
+            WHY THIS MATTERS:
+              This validation task runs all the tasks marked as configuration cacheable in the allow list,
+              to ensure the allow list contains only cacheable tasks.
+              Regular CI builds may mask configuration cache issues when non-cacheable
+              tasks disable the configuration cache.
+
+            HOW TO FIX (only if you have introduced a task, or upgraded a plugin):
+
+              1. Review the %s for specific issues
+
+              2. Common fixes for configuration cache problems:
+                  • Task.project at execution → Inject services (ProjectLayout, FileSystemOperations)
+                  • External processes → Use ProviderFactory.exec() or GradleExec (https://github.com/palantir/gradle-utils?tab=readme-ov-file#gradleexec)
+                  • Cannot serialize Gradle model types → Don't pass in full object use Property<T> for specific values or inject services
+
+              3. If you upgraded a plugin, verify it supports configuration cache
+
+              📚 Gradle Guide: https://github.com/palantir/gradle-guide/blob/develop/guide/adopting-the-configuration-cache.md
+              📚 Gradle docs: https://docs.gradle.org/current/userguide/configuration_cache.html
+            """,
+                reportsSection,
+                configCacheReportUrl.isPresent() || configCacheReportPath.isPresent()
+                        ? "Gradle configuration cache report"
+                        : "output above");
+
+        // Add validation output for local development only
+        if (validationReportUrl.isEmpty() && outputContent != null && !outputContent.isEmpty()) {
```

Comment:
`outputContent` is an `Optional` and should never be `null`, we should remove this check

similarly we should just use `isPresent` rather than inverting `isEmpty` on an `Optional`.